decidim-core/app/controllers/decidim/participatory_processes_controller.rb

```diff
@@ -7,14 +7,26 @@ module Decidim
   class ParticipatoryProcessesController < ApplicationController
     helper_method :participatory_processes, :participatory_process, :promoted_processes
 
+    def show
+      raise ActionController::RoutingError, "Not Found" unless participatory_process
+    end
+
     private
 
     def participatory_process
-      @participatory_process ||= participatory_processes.find(params[:id])
+      @participatory_process ||= if current_user && current_user.role?(:admin)
```

Comment:
Maybe we should consider using `Rectify::Query` objects: https://github.com/andypike/rectify#query-objects